notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * This file leverages Modal Bootstrap to display a single notice, like an Term of Service
+ */
+
+var upmodal_notice = upmodal_notice || {};
+
+if (!upmodal_notice.init) {
```

Comment:
:question: a portlet should only be initialized once, this appears to be guarding against multiple initializations. What is the expected use case for this guard?

Reply:
@ChristianMurphy -- This isn't portlet code, it's JS in the UI.  This file could be linked into the DOM multiple times, if the portlet appears on the page multiple times.

Reply:
So, this uses global variables, to create a singleton.
From my reading of the code, whichever instance starts first (and creates the singleton)  would work, the rest would do nothing.

Reply:
Adopting approach from other JS in this project.

The first time through the instance is initialized. Other instances would reference the first.